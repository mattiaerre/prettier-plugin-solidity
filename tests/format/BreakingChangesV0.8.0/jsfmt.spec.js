run_spec(__dirname, ['solidity-parse']);
run_spec(__dirname, ['solidity-parse'], { explicitTypes: 'never' });
run_spec(__dirname, ['solidity-parse'], { explicitTypes: 'preserve' });
run_spec(__dirname, ['solidity-parse'], {
  compiler: '0.8.0'
});
run_spec(__dirname, ['solidity-parse'], {
  compiler: 'earliest',
  explicitTypes: 'always'
});
run_spec(__dirname, ['solidity-parse'], {
  compiler: 'earliest',
  explicitTypes: 'never'
});
