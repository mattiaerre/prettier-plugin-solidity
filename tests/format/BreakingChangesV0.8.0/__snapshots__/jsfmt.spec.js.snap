// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreakingChangesV0.8.0.sol - {"compiler":"0.8.0"} format 1`] = `
====================================options=====================================
compiler: "0.8.0"
parsers: ["solidity-parse"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
// SPDX-License-Identifier: MIT
pragma solidity >=0.7.0 <=0.8.5;

contract VariableTypesMixed {
    bytes1 public c;
    byte public g;

    struct S {
        bytes1 c;
        byte g;
    }

    event Event(bytes1 _c, byte _g);

    function func(
        bytes1 _c,
        byte _g
    )
        public
        returns (
            bytes1,
            byte
        )
    {
        emit Event(_c, _g);
        return (_c, _g);
    }
}

=====================================output=====================================
// SPDX-License-Identifier: MIT
pragma solidity >=0.7.0 <=0.8.5;

contract VariableTypesMixed {
    bytes1 public c;
    bytes1 public g;

    struct S {
        bytes1 c;
        bytes1 g;
    }

    event Event(bytes1 _c, bytes1 _g);

    function func(bytes1 _c, bytes1 _g) public returns (bytes1, bytes1) {
        emit Event(_c, _g);
        return (_c, _g);
    }
}

================================================================================
`;

exports[`BreakingChangesV0.8.0.sol - {"compiler":"earliest","explicitTypes":"always"} format 1`] = `
====================================options=====================================
compiler: "earliest"
explicitTypes: "always"
parsers: ["solidity-parse"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
// SPDX-License-Identifier: MIT
pragma solidity >=0.7.0 <=0.8.5;

contract VariableTypesMixed {
    bytes1 public c;
    byte public g;

    struct S {
        bytes1 c;
        byte g;
    }

    event Event(bytes1 _c, byte _g);

    function func(
        bytes1 _c,
        byte _g
    )
        public
        returns (
            bytes1,
            byte
        )
    {
        emit Event(_c, _g);
        return (_c, _g);
    }
}

=====================================output=====================================
// SPDX-License-Identifier: MIT
pragma solidity >=0.7.0 <=0.8.5;

contract VariableTypesMixed {
    bytes1 public c;
    bytes1 public g;

    struct S {
        bytes1 c;
        bytes1 g;
    }

    event Event(bytes1 _c, bytes1 _g);

    function func(bytes1 _c, bytes1 _g) public returns (bytes1, bytes1) {
        emit Event(_c, _g);
        return (_c, _g);
    }
}

================================================================================
`;

exports[`BreakingChangesV0.8.0.sol - {"compiler":"earliest","explicitTypes":"never"} format 1`] = `
====================================options=====================================
compiler: "earliest"
explicitTypes: "never"
parsers: ["solidity-parse"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
// SPDX-License-Identifier: MIT
pragma solidity >=0.7.0 <=0.8.5;

contract VariableTypesMixed {
    bytes1 public c;
    byte public g;

    struct S {
        bytes1 c;
        byte g;
    }

    event Event(bytes1 _c, byte _g);

    function func(
        bytes1 _c,
        byte _g
    )
        public
        returns (
            bytes1,
            byte
        )
    {
        emit Event(_c, _g);
        return (_c, _g);
    }
}

=====================================output=====================================
// SPDX-License-Identifier: MIT
pragma solidity >=0.7.0 <=0.8.5;

contract VariableTypesMixed {
    byte public c;
    byte public g;

    struct S {
        byte c;
        byte g;
    }

    event Event(byte _c, byte _g);

    function func(byte _c, byte _g) public returns (byte, byte) {
        emit Event(_c, _g);
        return (_c, _g);
    }
}

================================================================================
`;

exports[`BreakingChangesV0.8.0.sol - {"explicitTypes":"never"} format 1`] = `
====================================options=====================================
explicitTypes: "never"
parsers: ["solidity-parse"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
// SPDX-License-Identifier: MIT
pragma solidity >=0.7.0 <=0.8.5;

contract VariableTypesMixed {
    bytes1 public c;
    byte public g;

    struct S {
        bytes1 c;
        byte g;
    }

    event Event(bytes1 _c, byte _g);

    function func(
        bytes1 _c,
        byte _g
    )
        public
        returns (
            bytes1,
            byte
        )
    {
        emit Event(_c, _g);
        return (_c, _g);
    }
}

=====================================output=====================================
// SPDX-License-Identifier: MIT
pragma solidity >=0.7.0 <=0.8.5;

contract VariableTypesMixed {
    bytes1 public c;
    bytes1 public g;

    struct S {
        bytes1 c;
        bytes1 g;
    }

    event Event(bytes1 _c, bytes1 _g);

    function func(bytes1 _c, bytes1 _g) public returns (bytes1, bytes1) {
        emit Event(_c, _g);
        return (_c, _g);
    }
}

================================================================================
`;

exports[`BreakingChangesV0.8.0.sol - {"explicitTypes":"preserve"} format 1`] = `
====================================options=====================================
explicitTypes: "preserve"
parsers: ["solidity-parse"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
// SPDX-License-Identifier: MIT
pragma solidity >=0.7.0 <=0.8.5;

contract VariableTypesMixed {
    bytes1 public c;
    byte public g;

    struct S {
        bytes1 c;
        byte g;
    }

    event Event(bytes1 _c, byte _g);

    function func(
        bytes1 _c,
        byte _g
    )
        public
        returns (
            bytes1,
            byte
        )
    {
        emit Event(_c, _g);
        return (_c, _g);
    }
}

=====================================output=====================================
// SPDX-License-Identifier: MIT
pragma solidity >=0.7.0 <=0.8.5;

contract VariableTypesMixed {
    bytes1 public c;
    bytes1 public g;

    struct S {
        bytes1 c;
        bytes1 g;
    }

    event Event(bytes1 _c, bytes1 _g);

    function func(bytes1 _c, bytes1 _g) public returns (bytes1, bytes1) {
        emit Event(_c, _g);
        return (_c, _g);
    }
}

================================================================================
`;

exports[`BreakingChangesV0.8.0.sol format 1`] = `
====================================options=====================================
parsers: ["solidity-parse"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
// SPDX-License-Identifier: MIT
pragma solidity >=0.7.0 <=0.8.5;

contract VariableTypesMixed {
    bytes1 public c;
    byte public g;

    struct S {
        bytes1 c;
        byte g;
    }

    event Event(bytes1 _c, byte _g);

    function func(
        bytes1 _c,
        byte _g
    )
        public
        returns (
            bytes1,
            byte
        )
    {
        emit Event(_c, _g);
        return (_c, _g);
    }
}

=====================================output=====================================
// SPDX-License-Identifier: MIT
pragma solidity >=0.7.0 <=0.8.5;

contract VariableTypesMixed {
    bytes1 public c;
    bytes1 public g;

    struct S {
        bytes1 c;
        bytes1 g;
    }

    event Event(bytes1 _c, bytes1 _g);

    function func(bytes1 _c, bytes1 _g) public returns (bytes1, bytes1) {
        emit Event(_c, _g);
        return (_c, _g);
    }
}

================================================================================
`;
